### PR TITLE
Task/290/parse amount and date columns

### PR DESCRIPTION
**Pull request by cyberbikepunk**
@akariv @roll

## Overview

This PR is about writing a caster processor that can be applied by default to all date and numbers in the datasets. The processor is [here](https://github.com/os-data/eu-structural-funds/blob/ab1d3bdbad18dd52aced805c1877e27e151c436a/common/processors/sniff_and_cast.py). Tests are [here](https://github.com/os-data/eu-structural-funds/blob/ab1d3bdbad18dd52aced805c1877e27e151c436a/tests/processors/test_sniff_and_cast.py).

### Assumptions

At this stage we assume that there is only one data resource. 

### Data processing

The processor goes through all fields in the schema and assigns them a caster. If the field has a complete set of format parameters, a ready made caster is used, else the processor sniffs the data to find the best match. If the processor fails to sniff a caster, a `CasterNotFound` error is raised.

Dates require the `format` parameter. Numbers require the `groupChar` and `decimalChar` parameters. Parameters can be set in the datapackage.

### Datapackage mutation

The processor toggles field types to match changes in the data.

## Remarks

- There is still a failing test relating to bad behavior in the `NumberType` class. I pushed a [PR](https://github.com/frictionlessdata/jsontableschema-py/pull/137).
- The PR also includes one pipeline as an example, which required a little tweaking of the xlsx ingestor
- There is still a small merge conflict I need to resolve 
- @roll please tell me if I re-invented the wheel. I'm _slowly_ trying to leverage more of your tooling. 
- @akariv The processor doesn't spew out any stats yet, mainly because I have to add a little stats collector to `utilities.process` first.
- There's no real reason why the constraint of a single resource couldn't be relaxed

### CODE REVIEW

File: common/bootstrap.py [hunk 1]
```diff
@@ -517,6 +574,42 @@ def validate_descriptions(ctx):
         secho(message, **color)
 
 
+@command(name='pipeline')
```
Comment (akariv) [inline common/bootstrap.py:577]: what's the use case for this command?

File: common/processors/concatenate_all_pipelines.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
+"""This processor concatenates all datasets into one."""
```
Comment (akariv) [inline common/processors/concatenate_all_pipelines.py:1]: why was this file resurrected?

**Comment by akariv** [comment]
Hey @cyberbikepunk 

Generally speaking I see no issues with the code (wrt bugs, style issues etc.)
I do think that there's some re-implementation of what already exists in the framework, but that's probably just the result of legacy code and lack of time to refactor.

File: common/bootstrap.py (hunk 1 above)
Comment (cyberbikepunk) [reply common/bootstrap.py:577]: the command manipulates `pipeline-spec` files in bulk. usage = 

```bootstrap --select key value pipeline insert processor_x --after processor_y``` 

I used it to bulk insert the new processor.

File: common/processors/concatenate_all_pipelines.py (hunk 2 above)
Comment (cyberbikepunk) [reply common/processors/concatenate_all_pipelines.py:1]: oops that came back with the merge conflict resolution. It should have disappeared. my bad.

**Comment by cyberbikepunk** [comment]
@akariv I think a guided tour of the framework is long overdue. i don't particularly relish re-inventing the wheel

**Comment by cyberbikepunk** [comment]
@akariv can you be more specific?

> I do think that there's some re-implementation of what already exists in the framework

**Comment by cyberbikepunk** [comment]
Solves #303

**Pull request merged**